app/javascript/components/ingredients/base-component.vue

```diff
@@ -77,39 +86,111 @@
 
 <script>
 
+import { getIngredients, postIngredient, deleteIngredient, editIngredient } from './../../api/ingredients.js';
+
 export default {
-  props: {
-    ingredients: { type: Array, required: true },
-  },
+
   data() {
     return {
       showingAdd: false,
       showingEdit: false,
       showingDel: false,
       ingredientToEdit: {},
+      ingredientToDelete: {},
       tableHeader: ['name', 'price', 'quantity', 'measure'],
+      ingredients: [],
+      status: '',
+      error: '',
     };
   },
 
+  async created() {
+    const token = localStorage.getItem('token');
+    const email = localStorage.getItem('email');
+
+    try {
+      const response = await getIngredients(email, token);
+      this.ingredients = response.data.data.map((element) => ({
+        id: element.id,
+        ...element.attributes,
+      }));
+      this.successResponse(response);
+    } catch (error) {
+      this.errorResponse(error);
+    }
+  },
+
   methods: {
     toggleAddModal() {
       this.showingAdd = !this.showingAdd;
     },
+
     toggleEditModal(ingredient) {
       this.showingEdit = !this.showingEdit;
       this.ingredientToEdit = ingredient;
     },
-    toggleDelModal() {
+
+    toggleDelModal(ingredient) {
       this.showingDel = !this.showingDel;
+      this.ingredientToDelete = ingredient;
     },
-    addIngredient() {
+
+    async addIngredient() {
+      const token = localStorage.getItem('token');
+      const email = localStorage.getItem('email');
       this.showingAdd = !this.showingAdd;
+
+      try {
+        const response = await postIngredient(this.$refs.addIngredientInfo.form, email, token);
+        const ingredientToAdd = { id: response.data.data.id, ...response.data.data.attributes };
```

Comment:
Ocupando [destructuring](https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/) esto podria quedar algo mas simple (en mi opinion)

```js
const {
  data:
    { data: { id, attributes}
  }
} = await postIngredient(this.$refs.addIngredientInfo.form, email, token);
const ingredientToAdd = { id, ...attributes };
```